scipy/_lib/_array_api.py

```diff
@@ -354,3 +357,17 @@ def xp_unsupported_param_msg(param):
 
 def is_complex(x, xp):
     return xp.isdtype(x.dtype, 'complex floating')
+
+def get_xp_devices(xp):
+    """Returns a list of available devices for the given namespace."""
+    if is_torch(xp):
+        devices = ['cpu']
+        import torch  # type: ignore[import]
+        num_cuda = torch.cuda.device_count()
+        for i in range(0, num_cuda):
+            devices += [f'cuda:{i}']
+        if torch.backends.mps.is_available():
+            devices += ['mps']
+        return devices
+
+    return [None]
```

Comment:
I think I need to add a similar case for `is_cupy`, forgot about that (I don't have multiple GPUs :upside_down_face: )

Reply:
Ah, CuPy haven't implemented the `device` parameter yet, so `array_api_compat.cupy` only accepts `'cpu'` or `None`. Not sure whether it is better to implement the device support in this helper now and skip the test, or just postpone CuPy in this helper.

Reply:
Hey @leofang , is there a rough ETA for CuPy implementing https://github.com/numpy/numpy/pull/25233?

Reply:
Oops, I meant specifically the fft commit, https://github.com/numpy/numpy/commit/6c510008a2515585d818e260c95343861c7cf761.

Reply:
I will implement `is_cupy` once I figure out how